lib/spack/spack/util/environment.py

```diff
@@ -528,13 +528,18 @@ def reversed(self):
 
         return rev
 
-    def apply_modifications(self):
+    def apply_modifications(self, env=None):
         """Applies the modifications and clears the list."""
+        # Use os.environ if not specified
+        # Do not copy, we want to modify it in place
```

Comment:
Making this into a `@contextmanager` would allow for recovering the original environment. I don't know if that's our goal: https://github.com/pantsbuild/pex/blob/f7eb554ecdbfdb9cd24a793b4dda358035e26855/pex/variables.py#L260-L268
```python
    @contextmanager
    def patch(self, **kw):
        # type: (**str) -> Iterator[Dict[str, str]]
        """Update the environment for the duration of a context."""
        old_environ = self._environ
        self._environ = self._environ.copy()
        self._environ.update(kw)
        yield self._environ
        self._environ = old_environ
```

This is not a blocking concern, just wanted to note an alternative approach.

Reply:
We have context managers that we can layer on top of this, but in many contexts (like forked processes) that's just overhead we don't want to maintain.

In this PR for example, I'm actually constructing an environment that gets passed to `subprocess.Popen`, and is never applied in the current process space.

Reply:
Thanks! That makes sense.